caluma/caluma_workflow/serializers.py

```diff
@@ -274,16 +311,30 @@ def validate(self, data):
     def update(self, instance, validated_data):
         instance = super().update(instance, validated_data)
         user = self.context["request"].user
-        instance.work_items.exclude(
+
+        for work_item in instance.work_items.exclude(
             status__in=[
                 models.WorkItem.STATUS_COMPLETED,
                 models.WorkItem.STATUS_CANCELED,
             ]
-        ).update(
-            status=models.WorkItem.STATUS_CANCELED,
-            closed_at=timezone.now(),
-            closed_by_user=user.username,
-            closed_by_group=user.group,
+        ):
+            work_item.status = models.WorkItem.STATUS_CANCELED
+            work_item.closed_at = timezone.now()
+            work_item.closed_by_user = user.username
+            work_item.closed_by_group = user.group
+            work_item.save()
+            send_event(
```

Comment:
I'm wondering if we should first update all work items, and only after the update send out the events. Otherwise, if there's some inter-relationship between the work items, the event handler may see some items in the old state, and some in the new

Reply:
Nice catch! Done.